c_src/libSigPyProcSpec.cpp

```diff
@@ -0,0 +1,249 @@
+#include <cmath>
+#include <cstring>
+#include <fftw3.h>
+#include <complex.h>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+
+#include <libUtils.hpp>
+
+namespace py = pybind11;
+
+void ccfft(py::array_t<float> inarray, py::array_t<float> outarray, int size) {
```

Comment:
This would be an API change, but you can create and return a numpy array, e.g:

```cpp
py::array_t<float> ccfft(py::array_t<float> inarray) {
    py::buffer_info inbuf = inarray.request();
    int nbytes = inbuf.size;

    auto outarray = py::array_t<float>(nbytes);
    py::buffer_info outbuf = outarray.request();

    float* indata  = (float*)inbuf.ptr;
    float* outdata = (float*)outbuf.ptr;

    fftwf_plan    plan;
    plan = fftwf_plan_dft_1d(size,
                             (fftwf_complex*)indata,
                             (fftwf_complex*)outdata,
                             FFTW_BACKWARD,
                             FFTW_ESTIMATE);
    fftwf_execute(plan);
    fftwf_destroy_plan(plan);
    
    return outarray;
}
```

If doing multiple FFTs the original API is still probably better (you're not creating a new numpy array each time), but this would be easier to use.

Reply:
TODO #8